code/game/objects/structures/nest.dm

```diff
@@ -0,0 +1,66 @@
+// Nests spawn mobs and objects when crossed by players.
+// Nests placed in "spawn_trigger_distance" will always spawn the same type of mob.
+
+/obj/structure/nest
+	name = "tunnel"
+	desc = "A twisted, dark passage to the underground."
+	icon = 'icons/mob/nest.dmi'
+	icon_state = "hole"
+
+	move_resist = INFINITY
+	anchored = TRUE
+	density = FALSE
+
+	var/faction = list("hostile")
+	var/spawn_byproduct = list(/obj/item/stack/ore/glass, /obj/item/stack/ore/iron)
+	var/spawn_byproduct_max = 3
+	var/spawn_is_triggered = FALSE
+	var/spawn_max = 2
+	var/spawn_mob_options = list(/mob/living/simple_animal/crab)
+	var/spawn_trigger_distance = 7
+
+/obj/structure/nest/examine(mob/user)
+	. = ..()
+	if(!spawn_is_triggered)
+		. += "You can hear a cacophony of growling snores from within."
```

Comment:
Maybe add a different span type so this sticks out, like `<span class='warning'>`

Reply:
Done in 2a0feef8cf9e1f432aa7ed8ee8945859d1c740a3!

![image](https://user-images.githubusercontent.com/33333517/180508793-93ea242b-b1b7-46d6-a76f-9a69f2d7a34c.png)